invoice-service/src/main/java/com/registry/invoiceservice/service/InvoiceService.java

```diff
@@ -65,12 +75,15 @@ public InvoiceDTO createInvoice(InvoiceDTO invoiceDTO) {
     }
 
     public InvoiceDTO getInvoiceById(Long id) {
-        return mapToInvoiceDTO(invoiceRepository.findById(id).get());
+        return mapToInvoiceDTO(invoiceRepository.findById(id).orElseThrow(() -> new InvoiceNotFoundException(String.valueOf(id))));
     }
 
-    public List<InvoiceDTO> getInvoicesByCriteria(InvoiceSearchRequestDTO criteriaDto) {
+    public List<InvoiceDTO> getInvoicesByCriteria(InvoiceSearchRequestDTO criteriaDto) throws InvoiceSearchException {
         List<InvoiceDTO> invoiceDTOS = new ArrayList<>();
         List<Invoice> invoices = invoiceRepository.searchInvoiceByCriteria(criteriaDto.getQuantityGreaterThan(), criteriaDto.getTotalPriceGreaterThan());
+        if(invoices.size() == 0){
```

Comment:
Having 0 results from a search is a valid case. If you did this to protect against NullPointerExceptions, look for other ways, like having your next operations depend on whether we have 0 or more results.